src/twisted/mail/test/test_pop3.py

```diff
@@ -119,93 +119,174 @@ def test_UIDListLineFormatter(self):
 
         listLines = list(pop3.formatUIDListResponse([0, None, 591],
-                         UIDs.__getitem__))
+                         uids.__getitem__))
         self.assertEqual(
             listLines,
             [b'+OK \r\n', b'1 abc\r\n', b'3 ghi\r\n', b'.\r\n'])
 
 
 
 class MyVirtualPOP3(mail.protocols.VirtualPOP3):
-
+    """
+    A virtual-domain-supporting POP3 server.
+    """
     magic = b'<moshez>'
 
     def authenticateUserAPOP(self, user, digest):
+        """
+        Authenticate against a user against a virtual domain.
+
+        @param user: The username.
+        @param digest: The digested password.
+
+        @return: A three-tuple like the one returned by
+            L{IRealm.requestAvatar}.  The mailbox will be for the user given
+            by C{user}.
+        """
         user, domain = self.lookupDomain(user)
         return self.service.domains[b'baz.com'].authenticateUserAPOP(
             user, digest, self.magic, domain)
 
 
 
 class DummyDomain:
+    """
+    A virtual domain for a POP3 server.
+    """
 
     def __init__(self):
         self.users = {}
 
 
     def addUser(self, name):
+        """
+        Create a mailbox for a new user.
+
+        @param name: The username.
+        """
         self.users[name] = []
 
 
     def addMessage(self, name, message):
+        """
+        Add a message to the mailbox of the named user.
+
+        @param name: The username.
+        @param message: The contents of the message.
+        """
         self.users[name].append(message)
 
 
     def authenticateUserAPOP(self, name, digest, magic, domain):
+        """
+        Succeed with a L{ListMailbox}.
+
+        @param name: The name of the user authenticating.
+        @param digest: ignored
+        @param magic: ignored
+        @param domain: ignored
+
+        @return: A three-tuple like the one returned by
+            L{IRealm.requestAvatar}.  The mailbox will be for the user given
+            by C{name}.
+        """
         return pop3.IMailbox, ListMailbox(self.users[name]), lambda: None
 
 
 
 class ListMailbox:
-
+    """
+    A simple in-memory list implementation of L{IMailbox}.
+    """
     def __init__(self, list):
+        """
+        @param list: The messages.
+        """
         self.list = list
 
 
     def listMessages(self, i=None):
+        """
+        Get some message information.
+
+        @param i: See L{pop3.IMailbox.listMessages}.
+        @return: See L{pop3.IMailbox.listMessages}.
+        """
         if i is None:
             return [len(l) for l in self.list]
         return len(self.list[i])
 
 
     def getMessage(self, i):
+        """
+        Get the message content.
+
+        @param i: See L{pop3.IMailbox.getMessage}.
+        @return: See L{pop3.IMailbox.getMessage}.
+        """
         return BytesIO(self.list[i])
 
 
     def getUidl(self, i):
+        """
+        Construct a UID which is simply the string representation of the given
```

Comment:
"Simply" is unnecessary and the return value isn't a string. It's an integer, but [the interface claims it should be `bytes`](https://github.com/twisted/twisted/blob/trunk/src/twisted/mail/interfaces.py#L185-L198); however, [the code now tolerates anything whose `str` representation can be encoded as UTF-8](https://github.com/twisted/twisted/blob/b64f480eed7a68672cf8f176d6d520c509c6097d/src/twisted/mail/pop3.py#L367-L368) , where it [used to tolerate anything that could be turned into a string](https://github.com/twisted/twisted/blob/c06d5c5e7/src/twisted/mail/pop3.py#L363).

Given all that, this should say something like "Return the given index as a UID."